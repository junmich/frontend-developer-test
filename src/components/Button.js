import React from 'react';

const Button = (props) => {
    let btnType = 'btn-default';
    switch(props.mode) {
        case 'primary':
            btnType = 'btn-primary';
            break;
        case 'info':
            btnType = 'btn-info';
            break;
       default:
            btnType = 'btn-default';
            break;

    }
    return <button
            className={`btn btn-lg ${btnType}`}
            onClick={props.onClick}
            type={props.type}
            disabled={props.disabled}
        >
            {props.name}
        </button>
}

Button.defaultProps = {
    name: "Button",
    mode: "default",
    type: "button",
    disabled: false,
    onClick: () => {}
}

export default Button;