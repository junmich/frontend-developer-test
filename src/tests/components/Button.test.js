import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../components/Button';

test('should render button correctly', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
});

// test('should render Input box correctly with props', () => {
//     const wrapper = shallow(<InputBox placeholder="Email" />);
//     expect(wrapper).toMatchSnapshot();
// });


// test('should render Input box correctly with props and set name', () => {
//     const wrapper = shallow(<InputBox placeholder="Email" name="email" />);
//     expect(wrapper).toMatchSnapshot();
// });