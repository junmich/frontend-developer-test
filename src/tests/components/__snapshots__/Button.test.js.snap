// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button correctly 1`] = `
<button
  className="btn btn-lg btn-default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;
